airflow/providers/google/cloud/utils/credentials_provider.py

```diff
@@ -261,6 +263,16 @@ def get_credentials_and_project(self) -> tuple[google.auth.credentials.Credentia
 
             project_id = _get_project_id_from_service_account_email(self.target_principal)
 
+        if self.lifetime:
+            # Create new impersonated credentials with an extended expiration time
+            credentials = impersonated_credentials.Credentials(
+                source_credentials=credentials.source_credentials,
```

Comment:
Regarding this comment on issue : https://github.com/apache/airflow/issues/31648#issuecomment-1572325675

I think it's because there are two different `Credentials` here

1. `Credentials` from `google.oauth2.service_account` [[source]](https://google-auth.readthedocs.io/en/master/_modules/google/oauth2/service_account.html#Credentials)
2. `Credentials` from `google.auth.impersonated_credentials` [[source]](https://google-auth.readthedocs.io/en/master/_modules/google/auth/impersonated_credentials.html#Credentials)

It looks like `credentials` could be one of both class according to `self.target_principal`